enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,502 @@
+
+#### Machine Config Operator
+
+Currently, MCO sets kubelet flags based on a [PlatformStatus and PlatformType](https://github.com/openshift/machine-config-operator/blob/7babc03f84088f49072c6f5b0d1769850d09a694/pkg/controller/template/render.go#L343).
+This flag is [crucial](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#running-cloud-controller-manager) for Cloud Controller Manager engagement within the cluster.
+
+Initially, the new "External" platform should be treated similarly to PlatformType "None" by the MCO, and do not set up any cloud-specific flags for the kubelet.
+Then, down the road (during [phase 3](#implementation-phases)), it would be expected for the MCO to use the "External" platform type and its spec as a signal about the underlying platform and cloud controller manager presence and operate accordingly.
+For an explicit signal about the necessity to set `--cloud-provider=external` flag to the kubelet, we will use the `CloudControllerManager` field of the `ExternalPlatformSpec`, which is  
+described in the [API Extensions](#api-extensions) section down below.
+
+#### Kube Controller Manager Operator
+
+Same as the [MCO](#machine-config-operator) in regard to kubelet, Kube Controller Manager Operator manages KCM (kube-controller-manager) deployments.
+
+Historically Kube Controller Manager was home for cloud-specific control loops. This logic is also engaging by setting up proper
+flags on KCM executable like,
+```shell
+...
+--cloud-provider=azure
+--cloud-config=/etc/kubernetes/cloud.conf
+...
+```
+
+For engaging an external Cloud Controller Manager no `cloud-provider` flag should be set for the KCM executable.
+
+In the context of this EP, no particular changes will be needed in the operator itself.
+
+#### Cluster Cloud Controller Manager Operator
+
+Responsible for deploying platform-specific Cloud Controller Manager as well as for handling a number of OpenShift's
+specific peculiarities (such as populating proxy settings for CCMs, sync credentials and so on).
+
+During the first phases of the "External" platform type enablement, this operator should be just disabled.
+In the future, when the delivery mechanism for CCMs will be defined, the operator might be engaged for deploying a user-provided cloud controller manager,
+however this is a subject for the upcoming design work. 
+
+#### Library Go
+
+Significant part of the code around PlatformType handling lives in the ["openshift/library-go"](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go).
+
+For the moment, this code is responsible for the decision around kubelet and kcm flags. Precisely, [IsCloudProviderExternal](https://github.com/openshift/library-go/blob/e1213f6ec5d10aa4aa8a4cac2780b1dc674c0396/pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go#L154) function
```

Comment:
I think I understand the statements in this section, but I'm not understanding what conclusion I should be taking away from it. What's the point here? Is this just identifying an important place in the code where changes will need to be made? Or is there a more significant point?

It may help to add one more sentence at the end of this section that draws a conclusion.

Reply:
I added another sentence. 

>Is this just identifying an important place in the code where changes will need to be made?
Basically yes. No another significant point here.